saleor/payment/providers/braintree/__init__.py

```diff
@@ -0,0 +1,258 @@
+import logging
+from decimal import Decimal
+from typing import Dict, List, Tuple
+
+import braintree as braintree_sdk
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.translation import pgettext_lazy
+from prices import Money
+
+from ... import TransactionType
+from ...models import Payment, Transaction
+from ...utils import create_transaction
+
+logger = logging.getLogger(__name__)
+# FIXME: Move to SiteSettings
+
+# If this option is checked, then one needs to authorize the amount paid
+# manually via the Braintree Dashboard
+CONFIRM_MANUALLY = False
+THREE_D_SECURE_REQUIRED = False
+
+# FIXME: Provide list of visible errors and messages translations
+# FIXME: We should also store universal visible errors for all payment
+# gateways, and parse gateway-specific errors to the unified version
+
+# Error codes whitelist should be a dict of code: error_msg_override
+# if no error_msg_override is provided,
+# then error message returned by the gateway will be used
+ERROR_CODES_WHITELIST = {
+    '91506': """
+        Cannot refund transaction unless it is settled.
+        Please try again later. Settlement time might vary depending
+        on the issuers bank.""",
+}
+
+
+def get_customer_data(payment: Payment) -> Dict:
+    return {
+        'billing': {
+            'first_name': payment.billing_first_name,
+            'last_name': payment.billing_last_name,
+            'company': payment.billing_company_name,
+            'postal_code': payment.billing_postal_code,
+            'street_address': payment.billing_address_1[:255],
+            'extended_address': payment.billing_address_2[:255],
+            "locality": payment.billing_city,
+            'region': payment.billing_country_area,
+            'country_code_alpha2': payment.billing_country_code},
+        'risk_data': {
+            'customer_ip': payment.customer_ip_address or ''},
+        'customer': {
+            'email': payment.billing_email}}
+
+
+def get_error_for_client(errors: List) -> str:
+    """Filters all error messages and decides which one is visible for the
+    client side.
+    """
+    if not errors:
+        return ''
+    default_msg = pgettext_lazy(
+        'payment error',
+        'Unable to process transaction. Please try again in a moment')
+    for error in errors:
```

Comment:
Could we log error to `stderr`? For example, I get this because I the DB is anonymized:
```python 
[{'code': '81813',
  'message': 'Postal code can only contain letters, numbers, spaces, and hyphens.'},
 {'code': '81813',
  'message': 'Postal code can only contain letters, numbers, spaces, and hyphens.'}]
```